jobs/kola-aws.Jenkinsfile

```diff
@@ -0,0 +1,67 @@
+@Library('github.com/coreos/coreos-ci-lib') _
+
+def streams
+node {
+    checkout scm
+    streams = load("streams.groovy")
+}
+
+properties([
+    pipelineTriggers([]),
+    parameters([
+      choice(name: 'STREAM',
+             // list devel first so that it's the default choice
+             choices: (streams.development + streams.production + streams.mechanical),
+             description: 'Fedora CoreOS stream to test'),
+      string(name: 'VERSION',
+             description: 'Fedora CoreOS Build ID to test',
+             defaultValue: '',
+             trim: true),
+      string(name: 'S3_STREAM_DIR',
+             description: 'Override the Fedora CoreOS S3 stream directory',
+             defaultValue: '',
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_IMAGE',
+             description: 'Override the coreos-assembler image to use',
+             defaultValue: "coreos-assembler:master",
+             trim: true)
+    ])
+])
+
+currentBuild.description = "[${params.STREAM}] - ${params.VERSION}"
+
+def s3_stream_dir = params.S3_STREAM_DIR
+if (s3_stream_dir == "") {
+    s3_stream_dir = "fcos-builds/prod/streams/${params.STREAM}"
+}
+
+cosaPod(image: params.COREOS_ASSEMBLER_IMAGE, memory: "256Mi",
+        secrets: ["aws-fcos-builds-bot-config", "aws-fcos-kola-bot-config"]) {
+
+    def AWS_FCOS_BUILDS_BOT_CONFIG = "/run/kubernetes/secrets/aws-fcos-builds-bot-config/config"
+    def AWS_FCOS_KOLA_BOT_CONFIG = "/run/kubernetes/secrets/aws-fcos-kola-bot-config/config"
```

Comment:
hmm. That's not exactly what I had in mind. I think I'm bumping into a limitation in my knowledge on how things work.

I was thinking we'd be somehow re-using 

https://github.com/coreos/fedora-coreos-pipeline/blob/35f1f18598cbe39601d71f5a181248288e6357bc/manifests/pod.yaml#L23-L24

rather than having to define it here again.

Reply:
We no longer use that pod definition in those jobs. Instead, we're telling `cosaPod` that we need access to the specific secrets (see https://github.com/coreos/coreos-ci-lib/pull/37). That way, pods only get the secrets they need mounted in.

I guess an improvement we could make here is to expose the path to the secret in environment variables as part of that wrapper.

Reply:
E.g. in this file, notice:

```groovy
cosaPod(image: params.COREOS_ASSEMBLER_IMAGE, memory: "256Mi",
        secrets: ["aws-fcos-builds-bot-config", "aws-fcos-kola-bot-config"]) {
```

Reply:
OK, sorry for the noise on this. I probably wouldn't have made the [other comment](https://github.com/coreos/fedora-coreos-pipeline/pull/270#discussion_r494999308) if I had realized how it worked.

Reply:
No worries, I think a good idea came out of it regardless: https://github.com/coreos/coreos-ci-lib/pull/39.